src/script/lua_api/l_object.cpp

```diff
@@ -1720,6 +1720,45 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+// set_time_offset(self, offset)
+int ObjectRef::l_set_time_offset(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == NULL)
+		return 0;
+
+	int offset_i = luaL_checknumber(L, 2);
+	u16 offset = offset_i;
+	//Negative values
+	if (offset_i<0) {
+		offset = 24000 + offset_i;
+	}
+
+	player->setTimeOffset(offset);
+	getServer(L)->SendTimeOfDay(player->getPeerId(), 
+		getEnv(L)->getTimeOfDay(), g_settings->getFloat("time_speed"));
+
+	lua_pushboolean(L, true);
+	return 1;
+}
+
+// get_time_offset(self)
+int ObjectRef::l_get_time_offset(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == NULL)
```

Comment:
`!player`

Reply:
Every other function in this file is with player == NULL too.